android/app-example/src/main/java/com/badoo/ribs/example/app/RootActivity.kt

```diff
@@ -0,0 +1,74 @@
+package com.badoo.ribs.example.app
+
+import android.os.Bundle
+import android.support.v7.app.AppCompatActivity
+import com.badoo.common.rib.BaseViewRouter
+import com.badoo.common.rib.directory.Directory
+import com.badoo.ribs.example.R
+import com.badoo.ribs.example.rib.foo_bar.FooBar
+import com.badoo.ribs.example.rib.foo_bar.builder.FooBarBuilder
+import io.reactivex.Observable
+import io.reactivex.ObservableSource
+import io.reactivex.functions.Consumer
+
+/** The sample app's single activity.  */
+class RootActivity : AppCompatActivity() {
+
+    private lateinit var rootRouter: BaseViewRouter<*>
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_root)
+        rootRouter = createRib().apply {
+            dispatchAttach(savedInstanceState)
+            onAttachToView(findViewById(R.id.root))
+        }
+    }
+
+    fun createRib(): BaseViewRouter<*> {
+        val rootBuilder =
+            FooBarBuilder(object : FooBar.Dependency {
+                override fun ribCustomisation(): Directory = AppRibCustomisations
```

Comment:
I'm would like to discuss and consider other ways for customization. Not in this example, but as a part of the architecture. I saw other examples of using that approach and I think we can improve the way of customization.

I remember @arkivanov also provided some cases when that approach no the best.

Reply:
Absolutely. It's on the agenda after MVP.